File: client-service/backend/client-service/src/main/java/com/cinema/clientservice/web/requests/HallSetupForRepertoire.java
package com.cinema.clientservice.web.requests;

import com.cinema.clientservice.web.dtos.SeatInfo;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.ArrayList;

public record HallSetupForRepertoire(
        String movieTitle,
        String languageVersion,
        Long repertoireId,
        Long hallId,
        LocalDate date,
        LocalDateTime startingTime,

        Integer numbOfRows,
        Integer numbOfSeatsInRow,
        ArrayList<ArrayList<SeatInfo>> seats
) {}

File: client-service/frontend/client-service/src/app/models/hallSetupforRepertoire.ts
import {SeatInfo} from "./seatInfo";

export interface HallSetupForRepertoire {
  movieTitle: string;
  languageVersion: string;
  repertoireId: number;
  hallId: number;
  date: Date;
  startingTime: Date;
  numbOfRows: number;
  numbOfSeatsInRow: number;
  seats: SeatInfo[][];
}
